lib/DDGC/GitHub.pm

```diff
@@ -8,25 +8,35 @@ use DateTime::Format::ISO8601;
 use DateTime::Duration;
 use HTTP::Request;
 use JSON::MaybeXS;
+use DDP;
 
-has ddgc => (
-  isa => 'DDGC',
-  is => 'ro',
-  weak_ref => 1,
-  required => 1,
-);
 
-sub gh { DDGC::GitHub::Cmds->new($_[0]->net_github->args_to_pass) }
+=head1 SYNOPSIS
 
-has net_github => (
-  is => 'ro',
-  lazy_build => 1,
-);
+    use DDGC::GitHub;
+
+    my $github = DDGC::GitHub->new(ddgc => $ddgc);
+
+    # update all the github_* databases
+    $github->update_database;
+
+    # something to do with users giving us access to their github accounts via oath
+    $github->validate_session_code($code, $user);
+
+    # find or update github_user table for this user
+    $github->find_or_update_user($login);
+
+=cut
+
+has ddgc       => (isa => 'DDGC', is => 'ro', weak_ref => 1, required => 1);
```

Comment:
@jbarrett I think you mentioned this to me as being a problem before.  Passing `$ddgc` around makes everything very tightly coupled.  How should I handle it?  

I think I should change this module ui to work like this 
```
my $github = DDGC::GitHub->new(
    db => $db,           # required; a DDGC::DB obj
);
```
and instantiate my own copy of DDGC::Config inside this module.  Ideally DDGC::Config would be singleton I guess?